### ISSUE TITLE
Uncaught TypeError: $(...).datepicker is not a function

### ISSUE THREAD

**Issue by guillosaracco** [jquery-ui-rails/jquery-ui-rails#101]
Hi there,
I've been trying and failing to implement the jQuery datepicker in my rails app for a while now.
Please let me know if I missed anything. Thanks in advance! This is what I did:

My rails version is '4.2.3'
My ruby version is '2.1.2'

I'm using
`ruby gem 'jquery-rails' '4.1.1'` (which shouldn't be a problem according to jquery-ui-rails readme)

1) I added:

``` ruby
gem 'jquery-ui-rails'
```

to my Gemfile, and did `bundle install`

2) I added `*= require jquery-ui/datepicker` in application.css.scss

3) I added `//= require jquery-ui/datepicker` in application.js

4) I added this textfield to a form in one of my views:

``` erb
<%= f.text_field :date, class: 'datepicker' %>
```

5) I added this in application.js

``` javascript
$(function() {
  $('.datepicker').datepicker();
});
```

**The problem:**
Now, after restarting the server and going to the view that contains the textfield two things happen:
1) The datepicker does not appear when clicking on the textfield
2) In the console I get this error: `Uncaught TypeError: $(...).datepicker is not a function`

Any ideas on what might be going on?
Thanks!

**Comment by kyleboe** [comment]
I am having a similar (and possibly related issue) when trying to use draggable.

Rails Version: `4.2.7.1`
Ruby Version: `2.3.1`

**Comment by jfredson** [comment]
I'm having the same problem. It looks like jquery-ui.js is just loading a blank file. Any ideas how to fix this?

Rails Version: 5.0.2
Ruby 2.3.1
jquery-ui-rails 6.0.1

**Comment by jfredson** [comment]
Ahh, I realize now that jquery-ui.js is just the master to load everything else from so it will show up blank in the browser anyway. I've verified that datepicker.js is loading fine, but I still get the error "Uncaught TypeError: $(...).datepicker is not a function". Am I somehow calling it wrong?

$('input.datepicker').datepicker();

**Comment by guillosaracco** [comment]
I was able to make this work a year ago, forgot about it, and now I encountered the same problem again.

Now I'm using:
Rails Version: 5.0.2
Ruby 2.3.1
jquery-ui-rails 6.0.1

This time, I got it working by rearranging the 'require' statements in application.js. 
I had to move the conflicting 'require' statement above ```//= require jquery-ui/widgets/datepicker```in order to make it work.

Hope it helps!
